src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Enemies from 'Parser/Core/Modules/Enemies';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+import {UNSTABLE_AFFLICTION_DEBUFF_IDS} from '../../Constants';
+
+class UABuffTracker extends Module {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  totalTicks = 0;
+  unbuffedTicks = 0;
+  ticksBuffedByReap = 0;
+  ticksBuffedByDrain = 0;
+  ticksBuffedByHaunt = 0;
+  ticksBuffedByBoth = 0;
+  hasMG = false;
+  hasHaunt = false;
+
+  on_initialized() {
+    if(!this.owner.error) {
+      this.hasMG = this.owner.selectedCombatant.hasTalent(SPELLS.MALEFIC_GRASP_TALENT.id);
+      this.hasHaunt = this.owner.selectedCombatant.hasTalent(SPELLS.HAUNT_TALENT.id);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if(UNSTABLE_AFFLICTION_DEBUFF_IDS.indexOf(spellId) === -1)
+      return;
+    const target = this.enemies.getEntity(event);
+    this.totalTicks++;
+    const buffedByReap = this.owner.selectedCombatant.hasBuff(SPELLS.DEADWIND_HARVESTER.id, event.timestamp);
+    const buffedByDrain = target.hasBuff(SPELLS.DRAIN_SOUL.id, event.timestamp);
+    const buffedByHaunt = target.hasBuff(SPELLS.HAUNT.id, event.timestamp);
+
+    if(this.hasMG) {
+      if(buffedByReap && buffedByDrain)
+        this.ticksBuffedByBoth++;
+      else if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else if(buffedByDrain)
+        this.ticksBuffedByDrain++;
+      else
+        this.unbuffedTicks++;
+    }
+    else if(this.hasHaunt) {
+      if(buffedByReap && buffedByHaunt)
+        this.ticksBuffedByBoth++;
+      else if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else if(buffedByHaunt)
+        this.ticksBuffedByHaunt++;
+      else
+        this.unbuffedTicks++;
+    }
+    else {
+      if(buffedByReap)
+        this.ticksBuffedByReap++;
+      else
+        this.unbuffedTicks++;
+    }
+  }
+
+  suggestions(when){
+    const buffPercentage = this.unbuffedTicks / this.totalTicks;
```

Comment:
technically this is an `unbuffedPercentage`, or even better `unbuffedTicksPercentage`. This naming had me confused initially (since I went straight to what looked to be the core of the suggestion, the isGreaterThan which would have been weird if this was some form of buffUptime which could have been named buffPercentage if you get what I mean).